client/app/components/parameters.js

```diff
@@ -42,6 +42,29 @@ function ParametersDirective($location) {
             scope.onUpdated();
           });
       };
+
+      scope.dirtyParamCount = 0;
+      scope.$watch(
+        'parameters',
+        () => {
+          scope.dirtyParamCount = size(filter(scope.parameters, 'hasPendingValue'));
+        },
+        true,
+      );
+
+      scope.isApplying = false;
+      scope.applyChanges = () => {
+        forEach(scope.parameters, p => p.applyPendingValue());
+        scope.isApplying = false;
+      };
+
+      scope.onApply = () => {
+        scope.isApplying = true;
+        scope.$apply();
+
+        scope.$apply(scope.applyChanges);
```

Comment:
The two `$apply` (one only for `scope.isApplying = true`) are to trigger the "updating" state?

Reply:
Yes. This part I'm not sure of since a) I don't know much Angular b) it applies so fast it's hard to notice the state change.

Reply:
:thinking:, this seems to provide the same result:

```javascript
scope.applyChanges = () => {
  scope.isApplying = true;
  forEach(scope.parameters, p => p.applyPendingValue());
  scope.isApplying = false;
};

scope.onApply = () => {
  scope.$apply(scope.applyChanges);
  scope.onValuesChange();
};
```

Reply:
Done 👍